sips/sip-x/sip-x-nft-metadata.md

```diff
@@ -0,0 +1,389 @@
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/
+This SIP’s copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+Tokens are digital assets registered on blockchain through a smart contract. A non-fungible token (NFT) is a token that is globally unique and can be identified through its unique identifier. In blockchains with smart contracts, including the Stacks blockchain, developers and users can use smart contracts to register and interact with non-fungible tokens.
+
+Some use cases of NFTs are name registration, digital art, certification, media and entertainment, real-estate. They all require that users associate certain content with an NFT. In general, it is helpful for the users to have a name, sound, image that represents this content.
+
+# Specification
+
+Every SIP-X compliant smart contract in Stacks blockchain must implement one or more functions that return a resolvable/retrievable uri referencing metadata. The metadata provide information e.g. for displaying a digital asset to users. This type of function is named "metadata uri functions".
+
+Appendix A contains a list of trait functions that must meet the following requirements for the return value. The appendix can be extended without changing the ratification status of this SIP.
+
+## Return Value of Metadata URI Functions
+
+The return value must be a `some` value if and only if the metadata reference an existing token, otherwise the value must be `none`. Appendix A specifies the exact meaning of "existing" for each function.
+
+For existing tokens, the inner value of the return value must be a string representing a resolvable URI.
+
+The schema of the resolvable URI is not specified and should be a well-known schema like `https`, `ar`, `ipfs`, `sia`. A `data` URI is also valid, however, the length is limited by this SIP.
+
+If a metadata uri function expects a parameter of type `uint` that identifies a token and the resulting strings contains `{id}`, then the `{id}` part must be replaced by the identifier in decimal format given in the function call.
+
+The resolved data of the URI must be a JSON blob.
+
+## JSON scheme of Metadata
+
+The JSON blob resolved through the uri must follow the following JSON schema.
+
+If metadata were retrieved by a function call containing a token identifier and the string `{id}` exists in any JSON value, it MUST be replaced with the actual token id in decimal format, by all client software that follows this standard.
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata",
+    "type": "object",
+    "required": ["version", "name"],
```

Comment:
isnt "image" required?

Reply:
I removed the requirement for image, because `properties.image_data` could be used or `image_data` directly. One of these three needs to be defined. I don't know how to specify that in the schema.